### ISSUE TITLE
Firmware issue

### ISSUE THREAD

**Issue by 4eef** [fabiobaltieri/usbasp-key#1]
Hello!
At the moment I need a linker which can link the hex via TPI interface used in ATtiny10. I have a FT232 linker but it cannot provide link via TPI (I'm using Windows). So I decided to make a USBasp linker, because it works fine with avrdude and has a TPI support with last firmware version.
Today I made your version of USBasp, using your PCB. I noticed that you've been used an old firmware version, which actually does not support linking via TPI interface. Have you updated your FW version to actual (2011-05-28)? Or, if not, please tell me, which things exactly was changed in your FW. I'll try to do changes to new FW myself.

![img_0088](https://cloud.githubusercontent.com/assets/10471137/5690049/57b7dd08-987d-11e4-9254-ba8e4e2d6db7.JPG)

**Comment by fabiobaltieri** [comment]
Hi! Yeah I never needed to use TPI so I never upgraded the base firmware to include support for that. It should be easy to port the latest release to the hardware since I started this branch from the plain 2009-10-23 firmware. The diff from the base commit is a bit messed up since there is a file conversion in the middle but it should be sufficient to reproduce this commits (the first 4 should be enough):

https://github.com/fabiobaltieri/usbasp-key/commit/e1b1b2193f1e0a9cb3c87c01657b12d8a2afb0b5
https://github.com/fabiobaltieri/usbasp-key/commit/d85b59fdc568937bbcd14ee972ed80d06a309ad8
https://github.com/fabiobaltieri/usbasp-key/commit/e5ba3d4779f53f2150b88546335cf4c19aaf9142
https://github.com/fabiobaltieri/usbasp-key/commit/46f3b817152c2837b01e39f571859f09a2612b5c
https://github.com/fabiobaltieri/usbasp-key/commit/1d6f70df7eaaf18ad91d8f58d7f3d3a0c65d8a11
https://github.com/fabiobaltieri/usbasp-key/commit/f61f34c66c5a9f1bc2096f2d5ab2d9d69f788adf

Nice prototype!

**Comment by 4eef** [comment]
Thanks a lot! There also needed pin remap for USB (see file usbconfig.h).
I've already compiled and programmed my own one and used it for linking the program to ATtiny10. Seems to be working perfect.
![nkcj8goei6g](https://cloud.githubusercontent.com/assets/10471137/5778407/c31f3d68-9da3-11e4-95ce-9ea09c3c42e3.jpg)
I've attached an archive with latest firmware version (2011-05-28) for your PCB in case of it needed by anyone.
http://rghost.ru/60385505
